### ISSUE TITLE
Custom IConnectionAdapter

### ISSUE THREAD

**Issue by thestonehead** [aspnet/KestrelHttpServer#2091]
Hello. 

I'm trying to implement a custom IConnectionAdapter and add it to ListenOptions, but a breakpoint inside OnConnectionAsync never gets hit. Am I doing something wrong?

```
public class Program
    {
        public static void Main(string[] args)
        {
            BuildWebHost(args).Run();
        }

        public static IWebHost BuildWebHost(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .UseKestrel(options =>
                {
                    options.Listen(System.Net.IPAddress.Any, 65187, listenOptions =>
                     {
                         listenOptions.ConnectionAdapters.Add(new Stuff());
                     });
                })
                .Build();
    }

    public class Stuff : Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal.IConnectionAdapter
    {
        public bool IsHttps => false;

        public Task<IAdaptedConnection> OnConnectionAsync(ConnectionAdapterContext context)
        {
            var adapted = new AdaptedConnection(context.ConnectionStream);
            return Task.FromResult<IAdaptedConnection>(adapted);
        }

        private class AdaptedConnection : IAdaptedConnection
        {
            public AdaptedConnection(Stream stream)
            {
                ConnectionStream = stream;
            }

            public Stream ConnectionStream { get; }

            public void Dispose()
            {
            }
        }
    }
```

The reason I'm trying to implement a custom IConnectionAdapter is to use HTTPS with short lived certificates where I would replace it at runtime with a new one before the old one expires.

**Comment by halter73** [comment]
@thestonehead Your sample code looks correct to me. OnConnectionAsync should get called. Are you sure you're making a request to port 65187?

Beware that you're using types in "Internal" namespaces which are subject to change even in minor version updates.

**Comment by thestonehead** [comment]
Ok, I've played around with it and emboldened by your validation, found out it was about the way I was running the project. When I ran it in Visual studio on IIS Express, it didn't work, but when I changed that it did. 
So... false alarm :D 

And yes, I'm aware they intend to deprecate Connection adapters in the next version and implement a better pipeline architecture. It's still the simplest way to achieve dynamic switching of short lived certificates.
